server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -666,16 +636,20 @@ public void run() {
    * balanceTablets() balances tables by DataLevel. Return the current set of migrations partitioned
    * by DataLevel
    */
-  private static Map<DataLevel,Set<KeyExtent>>
-      partitionMigrations(final Set<KeyExtent> migrations) {
+  private Map<DataLevel,Set<KeyExtent>> partitionMigrations() {
     final Map<DataLevel,Set<KeyExtent>> partitionedMigrations = new EnumMap<>(DataLevel.class);
-    // populate to prevent NPE
     for (DataLevel dl : DataLevel.values()) {
-      partitionedMigrations.put(dl, new HashSet<>());
+      Set<KeyExtent> extents = new HashSet<>();
+      // prev row needed for the extent
+      try (var tabletsMetadata = getContext().getAmple().readTablets().forLevel(dl)
+          .fetch(TabletMetadata.ColumnType.PREV_ROW, TabletMetadata.ColumnType.MIGRATION).build()) {
```

Comment:
Not a change for this PR, but these scans for migrations could be optimized.  Normally this will bring back a ton of prev row columns and a few migration columns.  We could do a simple server side filter or something more complex that uses two iterators one that scans on the migration column and a second iterator that is used to get the prev row column only when the migration is seen.